unlock-app/src/components/interface/checkout/main/Confirm/ConfirmCrossmint.tsx

```diff
@@ -0,0 +1,246 @@
+import {
+  CrossmintPaymentElement,
+  useCrossmintEvents,
+} from '@crossmint/client-sdk-react-ui'
+import { CheckoutService } from './../checkoutMachine'
+import { Connected } from '../../Connected'
+import { useConfig } from '~/utils/withConfig'
+import { Fragment, useState } from 'react'
+import { useActor } from '@xstate/react'
+import { PoweredByUnlock } from '../../PoweredByUnlock'
+import { Pricing } from '../../Lock'
+import { getReferrer, lockTickerSymbol } from '~/utils/checkoutLockUtils'
+import { Lock } from '~/unlockTypes'
+import { RiErrorWarningFill as ErrorIcon } from 'react-icons/ri'
+import { ViewContract } from '../../ViewContract'
+// import { useUpdateUsersMetadata } from '~/hooks/useUserMetadata'
+import { usePricing } from '~/hooks/usePricing'
+import { usePurchaseData } from '~/hooks/usePurchaseData'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { ethers } from 'ethers'
+import { useCrossmintEnabled } from '~/hooks/useCrossmintEnabled'
+import { TransactionAnimation } from '../../Shell'
+
+interface Props {
+  injectedProvider: unknown
+  checkoutService: CheckoutService
+  onConfirmed: (lock: string, hash?: string) => void
+  onError: (message: string) => void
+}
+
+interface CrossmintQuote {
+  lineItems: any[]
+  totalPrice: {
+    amount: string
+    currency: string
+  }
+}
+
+export function ConfirmCrossmint({
+  injectedProvider,
+  onConfirmed,
+  // onError,
+  checkoutService,
+}: Props) {
+  const { email } = useAuth()
+  const [state] = useActor(checkoutService)
+  const config = useConfig()
+  const [isConfirming, setIsConfirming] = useState(false)
+  const [quote, setQuote] = useState<CrossmintQuote | null>(null)
+  const crossmintEnv = config.env === 'prod' ? 'production' : 'staging'
+
+  const {
+    lock,
+    recipients,
+    paywallConfig,
+    // metadata, // Should we save it?
+    data,
+    keyManagers,
+  } = state.context
+
+  // const { mutateAsync: updateUsersMetadata } = useUpdateUsersMetadata()
+
+  const { isLoading: isCrossmintLoading, data: crossmintConfigId } =
+    useCrossmintEnabled({
+      network: lock!.network,
+      lockAddress: lock!.address,
+    })
+
+  const { isInitialLoading: isInitialDataLoading, data: purchaseData } =
+    usePurchaseData({
+      lockAddress: lock!.address,
+      network: lock!.network,
+      paywallConfig,
+      recipients,
+      data,
+    })
+
+  // Handling minting events
+  const { listenToMintingEvents } = useCrossmintEvents({
+    environment: crossmintEnv,
+  })
+
+  // Handling payment events
+  const onCrossmintPaymentEvent = (paymentEvent: any) => {
```

Comment:
Is there no better type than any here from the sdk library?

Reply:
no :/

Reply:
Actually there is ! updated!